tests/models/test_cpu.py

```diff
@@ -75,57 +70,85 @@ def test_cpu_slurm_save_load(tmpdir):
     # new logger file to get meta
     logger = tutils.get_default_logger(tmpdir, version=version)
 
+    model = BoringModel()
+
+    class _StartCallback(Callback):
+        def on_init_start(self, trainer):
+            print("Starting to init trainer!")
+
+        # set the epoch start hook so we can predict before the model does the full training
+        def on_epoch_start(self, trainer, model):
+            assert trainer.global_step == real_global_step and trainer.global_step > 0
+            # predict with loaded model to make sure answers are the same
+            model.eval()
+            new_pred = model(batch)
+            assert torch.all(torch.eq(pred_before_saving, new_pred)).item() == 1
+
     trainer = Trainer(
         default_root_dir=tmpdir,
         max_epochs=1,
         logger=logger,
-        checkpoint_callback=ModelCheckpoint(dirpath=tmpdir),
+        callbacks=[_StartCallback(), ModelCheckpoint(dirpath=tmpdir)],
     )
-    model = EvalModelTemplate(**hparams)
-
-    # set the epoch start hook so we can predict before the model does the full training
-    def assert_pred_same():
-        assert trainer.global_step == real_global_step and trainer.global_step > 0
-
-        # predict with loaded model to make sure answers are the same
-        trainer.model.eval()
-        new_pred = trainer.model(x)
-        assert torch.all(torch.eq(pred_before_saving, new_pred)).item() == 1
-
-    model.on_epoch_start = assert_pred_same
-
     # by calling fit again, we trigger training, loading weights from the cluster
     # and our hook to predict using current model before any more weight updates
     trainer.fit(model)
 
 
-def test_early_stopping_cpu_model(tmpdir):
+def test_all_features_cpu_model(tmpdir):
     """Test each of the trainer options."""
-    stopping = EarlyStopping(monitor='early_stop_on', min_delta=0.1)
```

Comment:
I think you need to add the `EarlyStopping` callback back here since it's not added by default. Merge the below test here itself?

Reply:
Hi @rohitgr7, thanks for the suggestion! However, I organized the code to tests callbacks separately from other trainer options to keep tests simpler and cleaner. Let me know if it sounds good :)

Reply:
All the callbacks are already tested in tests/callbacks. This test doesn't do anything special, just takes all the arguments even the ModelCheckpoint will be added by default. Only EarlyStopping is missing, so I think it's not necessary here to create a separate one.

Reply:
I see your point, sounds good to me :)